Sources/FoundationEssentials/Data/Data+Reading.swift

```diff
@@ -336,7 +336,11 @@ internal func readBytesFromFile(path inPath: PathOrURL, reportProgress: Bool, ma
     }
     
     let fileSize = min(Int(clamping: filestat.st_size), maxLength ?? Int.max)
+#if !os(Windows)
```

Comment:
Is there any way at all to structure this so that the `S_IFMT` is cast to a `mode_t` on all platforms, without an ambiguity warning or error?

This kind of ifdef is so unfortunate, because it complicates the code and produces a fork to maintain but has no difference in behavior at all.

Reply:
This code is already in the `#else` block of an `#if os(Windows)` statement above, so this code is already `#if !os(Windows)`. If we need to make this change for all non-windows platforms, I think we can just outright change the existing line since the existing line is already conditionalized

Reply:
The problem is that with Musl, the Clang importer successfully imports the underlying `S_IFMT` (as a `CInt`) and also sees the `S_IFMT` (a `mode_t`) from the overlay. Doing `mode_t(S_IFMT)` is then ambiguous because it doesn't know whether you meant the `CInt` or the `mode_t` constant.

I had thought that the explicit conversion was there because on Windows, the `S_IFMT` constant in the overlay is _not_ a `mode_t`, but you're right, that entire block of code is not used on Windows, so we can just remove the conversion. I'll update the PR.